sqlmesh/dbt/source.py

```diff
@@ -44,6 +44,7 @@ class SourceConfig(GeneralConfig):
     loaded_at_field: t.Optional[str] = None
     quoting: t.Dict[str, t.Optional[bool]] = {}
     external: t.Optional[t.Dict[str, t.Any]] = {}
+    source_meta: t.Optional[t.Dict[str, t.Any]] = {}
```

Comment:
Is there a reason why you had to do this vs using the existing `meta` field on the `GeneralConfig` base class?

Reply:
yes i considered this, but I noticed that in `dbt` they have both fields: [dbt source definition](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/artifacts/resources/v1/source_definition.py#L65).

where `source_meta` is the top-level meta key. for example when we define a source as:

```yaml
sources:
  - name: parquet
    meta:
      external_location: "s3://{name}.parquet"
    tables:
      - name: table name
        meta: 
          owner: "@Alice"
```

in dbt's `SourceDefinition` we would get:

```python
meta = {'owner': '@alice'}
source_meta = {'external_location': "s3://{name}.parquet"}
```

so I wanted to ensure that both these fields could be mapped in our `SourceConfig` to align with their definition

Reply:
Oooh I see, of course. Makes sense!